hathor/transaction/block.py

```diff
@@ -138,6 +140,17 @@ def calculate_height(self) -> int:
         parent_block = self.get_block_parent()
         return parent_block.get_metadata().height + 1
 
+    def calculate_min_height(self) -> int:
+        """Same as height for a block.
+        """
+        assert self.storage is not None
+        return max(chain(
+            # 1) height of parent block + 1
+            [self.calculate_height()],
```

Comment:
I got your logic. I don't like the structure itself but it might work. Anyway, why did you include the current height?